releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java

```diff
@@ -56,6 +56,10 @@ public final class NotesBuilder {
             " https://github.com/%s/issues/%d";
     /** String format pattern for warning if issue is not closed. */
     private static final String MESSAGE_ISSUE_NOT_FOUND = "[WARN] Issue #%d could not be found!";
+
+    /** String format pattern for warning if commit threw an exception during handling.  */
+    private static final String MESSAGE_PROBLEM_HANDLING_COMMIT = "[WARN] An exception occured "
+                            + "while handling commit \"%s\". Assigning miscellaneous label to it.";
```

Comment:
Where is test showing new message?

Reply:
There is no error thrown in the new tests.

Reply:
Yes, it is probably because of the change at https://github.com/checkstyle/contribution/pull/770#discussion_r1096573891 which @romani said was not wanted. I think that code has to be reverted, then your new message will appear.

Reply:
@stoyanK7, let me know if something needs to be clarified more.
This update blocks our ability to test release.

Reply:
Please make commit message like `Issue #asd: asd`

Reply:
On it.

Reply:
Aight, reverting back to `return message.startsWith("Issue") || message.startsWith("Pull");` creates an issue with handling all commits.

Specifically,
https://github.com/checkstyle/contribution/blob/9850b218700203e9c107add7b38a38c5cf7aff8d/releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java#L240-L251

when `commitMessage.getIssueNumber()` is called with let's say `Issue 11: Mock issue title 11 missing #`, it fails the following way since it assumes there is a number between the `#` and the `:`:

![Screenshot from 2023-02-08 09-49-33](https://user-images.githubusercontent.com/23459549/217480572-427b7b6e-c822-4967-9a31-b2b6f9d567d5.png)

Reply:
Same goes with  `Issue #asd: asd`. Wouldn't it be a good idea to change the implementation of `isIssueOrPull()` to use regex to prevent this and check for `^(Pull|Issue) #\d+:.*$`?

![Screenshot from 2023-02-08 10-19-46](https://user-images.githubusercontent.com/23459549/217487641-eed821c9-b7be-4307-8570-9cb2c6ed4d83.png)

Reply:
All calls to `getIssueNumber()` are guarded by checking if `isIssueOrPull()`.

Reply:
Yes, please boost isIssueOrPull() by all means.
Catching exception still good, to not deal with any unfortunate edge case

Reply:
@romani By above method, you mean `getCommitsForIssue()`?

Reply:
Done. Boosted implementation and added one more test case.